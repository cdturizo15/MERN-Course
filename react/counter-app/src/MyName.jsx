import { Fragment } from "react"
import PropTypes from 'prop-types'

// No se pueden renderizar objetos
// Las promesas son objetos
const MyName = ({name, description}) => {

    return (
    <div>
        <h1 data-testid='test-title'>My name is {name}</h1>
        <p>Soy un {description}</p>
    </div>      
  )
}

MyName.propTypes = {
  description: PropTypes.string.isRequired
}

MyName.defaultProps = {
  description: 'No hay descripcion'
}
export default MyName