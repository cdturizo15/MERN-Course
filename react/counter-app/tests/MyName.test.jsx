import { render } from "@testing-library/react"
import MyName from "../src/MyName"

 

describe('Pruebas MyName component', () => {
    test('debe hacer match con snapshot', () => {
        const {container} = render( <MyName name={'Cristian'}description={'crack'}  /> )
        expect(container).toMatchSnapshot();
    })
    test('Un h1 presenta tu nombre', () => {
        const name = 'My name is Cristian'
        const { getByTestId } = render( <MyName name={'Cristian'}description={'crack'}  /> )
        expect(getByTestId('test-title').innerHTML).toBe(name)
    })
})