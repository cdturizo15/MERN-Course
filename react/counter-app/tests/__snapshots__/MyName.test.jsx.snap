// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas MyName component debe hacer match con snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      My name is 
      Cristian
    </h1>
    <p>
      Soy un 
      crack
    </p>
  </div>
</div>
`;
